### ISSUE TITLE
Fix S3267 FN: Support list pattern

### ISSUE THREAD

**Issue by cristian-ambrosini-sonarsource** [SonarSource/sonar-dotnet#6387]
### Description

FN for S3267 (LoopsAndLinq).

Using C#11 list pattern, the rule is not raising an issue.

### Repro steps

```csharp
        void ListPattern(List<int[]> list)
        {
            foreach (int[] array in list) // FN
            {
                if (array is [1, 2, 3]) // FN
                {
                    Console.WriteLine("Test");
                }
            }
        }
```

### Expected behavior

The rule should raise an issue.

### Actual behavior

The rule does not raise an issue.
